parity/snapshot_cmd.rs

```diff
@@ -257,20 +258,25 @@ impl SnapshotCommand {
 		let writer = PackedWriter::new(&file_path)
 			.map_err(|e| format!("Failed to open snapshot writer: {}", e))?;
 
-		let progress = Arc::new(Progress::new());
+		let progress = Arc::new(RwLock::new(Progress::new()));
 		let p = progress.clone();
 		let informant_handle = ::std::thread::spawn(move || {
 			::std::thread::sleep(Duration::from_secs(5));
-
 			let mut last_size = 0;
-			while !p.done() {
-				let cur_size = p.bytes();
-				if cur_size != last_size {
-					last_size = cur_size;
-					let bytes = ::informant::format_bytes(cur_size as usize);
-					info!("Snapshot: {} accounts (state), {} blocks, {} bytes", p.accounts(), p.blocks(), bytes);
+			loop {
+				{
+					let progress = p.read();
+					if !progress.done() {
+						let cur_size = progress.bytes();
+						if cur_size != last_size {
+							last_size = cur_size;
+							let bytes = ::informant::format_bytes(cur_size as usize);
```

Comment:
this will truncate `u64` when `usize::max_value() < u64::max_value()`, I don't know it is issue or if we just should change `informant::format_bytes` to take an `u64` instead

Reply:
The `usize`s here come from the io methods deeper down the stack that actually write bytes; they all return `usize` for the nr of bytes written. I wanted to be consistent and went with `u64` but we can go the other way too if preferred. I don't have a strong opinion tbh.

Reply:
ok, I don't have strong opinions of this but I think `truncation casting` deserves a comment because it is not clear to me at least whether it is intended or bug.

Reply:
I went the other way in #ee4a0a9b0 ad changed `format_bytes()` to take a `u64` instead.